src/clr-angular/utils/a11y/aria-live.service.ts

```diff
@@ -0,0 +1,113 @@
+}
+
+/**
+ * Time in miliseconds before inserting the content into the contianer
+ */
+export const ARIA_LIVE_TICK: number = 100;
+
+@Injectable()
+export class AriaLiveService implements OnDestroy {
+  private ariaLiveElement: HTMLElement;
+  private document: Document;
+  private previousTimeout: number;
+  private id: string = `clr-aria-live-element-${uniqueIdFactory()}`;
+
+  constructor(
+    private _ngZone: NgZone,
+    @Inject(DOCUMENT) _document: any,
+    @Inject(PLATFORM_ID) private platformId: Object
+  ) {
+    this.document = _document;
+
+    if (isPlatformBrowser(platformId)) {
+      this.ariaLiveElement = this.createLiveAriaContainer();
+    }
+  }
+
+  private createLiveAriaContainer(): HTMLElement {
+    const ariaLiveElement = this.document.createElement('div');
+
+    ariaLiveElement.setAttribute('id', this.getId());
+    // Use clarity screen reader class to hide the dom element
+    // and fix the scrollbar shake
+    ariaLiveElement.classList.add('clr-sr-only');
+
+    ariaLiveElement.setAttribute('aria-atomic', 'true');
+    ariaLiveElement.setAttribute('aria-live', AriaLivePoliteness.polite);
+
+    this.document.body.appendChild(ariaLiveElement);
+
+    return ariaLiveElement;
+  }
+
+  public announce(message: string, politeness: AriaLivePoliteness = AriaLivePoliteness.polite): Promise<void> {
+    // Don't announce anything if there is no ariaLiveElement
+    if (!this.ariaLiveElement) {
+      return;
+    }
+
+    this.clear();
+    clearTimeout(this.previousTimeout);
+
+    this.ariaLiveElement.setAttribute('aria-live', politeness);
+
+    // This 100ms timeout is necessary for some browser + screen-reader combinations:
+    // - Both JAWS and NVDA over IE11 will not announce anything without a non-zero timeout.
+    // - With Chrome and IE11 with NVDA or JAWS, a repeated (identical) message won't be read a
+    //   second time without clearing and then using a non-zero delay.
+    // (using JAWS 17 at time of this writing).
+    return this._ngZone.runOutsideAngular(() => {
+      return new Promise(resolve => {
+        clearTimeout(this.previousTimeout);
+        /**
+         * Cause I didn't want to use `window.setTimeout` so TypeScript will stop throwing
+         * this error `Type 'Timer' is not assignable to type 'number'` , I had to use `<any>` before
+         * setTimeout()
+         *
+         * More on the topic here: https://github.com/TypeStrong/atom-typescript/issues/1053
+         */
+        this.previousTimeout = <any>setTimeout(() => {
+          this.ariaLiveElement.textContent = message;
+          resolve();
+        }, ARIA_LIVE_TICK);
+      });
+    });
+  }
+
+  /**
+   * Remove the content inside the AriaLive Container
+   */
+  public clear() {
+    if (this.ariaLiveElement) {
+      this.ariaLiveElement.textContent = '';
+    }
+  }
+
+  public ngOnDestroy() {
+    clearTimeout(this.previousTimeout);
+
+    if (this.ariaLiveElement && this.ariaLiveElement.parentNode) {
```

Comment:
We are still missing platform check here.

Reply:
We don't need the check for `this.ariaLiveElement.parentNode` any more and I will remove it.

For the platform check, don't see a reason to add another `&&` or `if` for that. There is a check inside the `constructor` that will prevent the creation of the `this.ariaLiveElement` and that is the only place where this variable is assigned. So the check for `undefined` will cover it.